src/ResourceManager/ContainerRegistry/Commands.ContainerRegistry/Commands/NewAzureContainerRegistryReplication.cs

```diff
@@ -0,0 +1,91 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.ResourceManager.Common.Tags;
+using System.Collections;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.ContainerRegistry
+{
+    [Cmdlet(VerbsCommon.New, ContainerRegistryReplicationNoun, SupportsShouldProcess = true)]
+    [OutputType(typeof(PSContainerRegistryReplication))]
+    public class NewAzureContainerRegistryReplication : ContainerRegistryCmdletBase
+    {
+        [Parameter(Position = 0, Mandatory = true, ParameterSetName = NameResourceGroupParameterSet, HelpMessage = "Resource Group Name.")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(Position = 1, Mandatory = true, ParameterSetName = NameResourceGroupParameterSet, HelpMessage = "Container Registry Name.")]
+        [Alias(ContainerRegistryNameAlias, ResourceNameAlias)]
+        [ValidateNotNullOrEmpty]
+        public string RegistryName { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = RegistryObjectParameterSet, HelpMessage = "Container Registry Object.")]
+        [ValidateNotNull]
+        public PSContainerRegistry Registry { get; set; }
+
+        [Parameter(Mandatory = true, HelpMessage = "Container Registry Location. Default to the location of the resource group.")]
+        [ValidateNotNullOrEmpty]
+        [Alias(ReplicationLocationAlias)]
+        public string Location { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Container Registry Replication Name. Default to the location name.")]
```

Comment:
@Jinhuafei any reason that you don't want the user to _always_ provide a `Name` of their choosing?

Reply:
The default name of the replication is the location of the registry, and we prefer to default name.

Reply:
@Jinhuafei out of curiosity, is that the current behavior on the portal?

Reply:
ACR portal doesn't allow users to enter replication name. Az CLI allows it for consistency reason but is optional.